rules/type-error.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const tcIdentifiers = new Set([
```

Comment:
As far as i know only `isNaN` and `isFinite` are affected. I added a check for global type checking identifiers, too. 👍